tetrad-gui/src/main/java/edu/cmu/tetradapp/ui/model/AlgorithmModels.java

```diff
@@ -69,13 +70,28 @@ public static AlgorithmModels getInstance() {
         return INSTANCE;
     }
 
-    public List<AlgorithmModel> getModels() {
-        return models;
+    private List<AlgorithmModel> filterInclusivelyByAllOrSpecificDataType(List<AlgorithmModel> algorithmModels, DataType dataType) {
+        return (dataType == DataType.All)
+                ? algorithmModels
+                : algorithmModels.stream()
+                        .filter(e -> {
+                            for (DataType dt : e.getAlgorithm().getAnnotation().dataType()) {
+                                if (!(dt == DataType.All || dt == dataType)) {
```

Comment:
I think the logic here in the for loop is not right. For an algorithm that can handle both continuous and discrete data, if the annotation has continuous specified before discrete, and the input data is discrete, this algorithm won't be included. For example. 

````
@edu.cmu.tetrad.annotation.Algorithm(
        name = "GFCI",
        command = "gfci",
        algoType = AlgType.allow_latent_common_causes,
        dataType = {DataType.Continuous, DataType.Discrete, DataType.Mixed}
````

GFCI won't show up in the algo chooser when I use some simulated discrete data, because the first data type is continuous and it will return false.

Reply:
Good catch!  Just fixed it.